### PR TITLE
fix: [CDS-34326]: nested template variables

### PR DESCRIPTION
**Pull request by yc147**
##### Jira Links:

https://harness.atlassian.net/browse/CDS-34326

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx [hunk 1]
```diff
@@ -91,9 +100,18 @@ describe('<TemplateVariables /> tests', () => {
...
-            value={{ variablesTemplate, loading: false, initLoading: true, error: null, metadataMap } as any}
+            value={
+              {
+                originalTemplate: template,
+                variablesTemplate,
+                loading: false,
+                initLoading: true,
+                error: null,
+                metadataMap
+              } as any
```
Comment (bhavya-sinha) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx:111]: Same

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx [hunk 2]
```diff
@@ -75,9 +75,18 @@ describe('<TemplateVariables /> tests', () => {
...
-            value={{ variablesTemplate, loading: false, initLoading: false, error: null, metadataMap } as any}
+            value={
+              {
+                originalTemplate: template,
+                variablesTemplate,
+                loading: false,
+                initLoading: false,
+                error: null,
+                metadataMap
+              } as any
```
[the remark this answers is not in the document]
Comment (yc147) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx]: Added

File: src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx (hunk 1 above)
Comment (yc147) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateVariables/_test_/TemplateVariables.test.tsx]: Added

**Review by bhavya-sinha — APPROVED** [review]

**Review by bhavya-sinha — DISMISSED** [review]

File: src/modules/70-pipeline/components/TemplateVariablesContext/TemplateVariablesContext.tsx [hunk 3]
```diff
@@ -55,20 +68,47 @@ export function TemplateVariablesContextProvider(
...
+    }
+  }, [resolvedTemplateResponse])
+
   React.useEffect(() => {
     setTemplateVariablesData({
       metadataMap: defaultTo(data?.data?.metadataMap, {}),
-      variablesTemplate: defaultTo(parse(defaultTo(data?.data?.yaml, '')), {})
+      variablesTemplate: get(parse(defaultTo(data?.data?.yaml, '')), resolvedTemplate.type.toLowerCase())
     })
   }, [data?.data?.metadataMap, data?.data?.yaml])
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/TemplateVariablesContext/TemplateVariablesContext.tsx:101]: do we need to add resolvedTemplate as a dependency here ?

**Review by samarth-sharma31 — APPROVED** [review]

File: src/modules/70-pipeline/components/TemplateVariablesContext/TemplateVariablesContext.tsx (hunk 3 above)
Comment (yc147) [reply src/modules/70-pipeline/components/TemplateVariablesContext/TemplateVariablesContext.tsx:101]: `resolvedTemplate.type` will never change

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**